encoders/nlp/TransformerTorchEncoder/tests/test_transformertorchencoder.py

```diff
@@ -1,140 +1,144 @@
-    },
-    {
-        "pooling_strategy": 'auto',
-        "pretrained_model_name_or_path": 'distilbert-base-uncased',
-        "model_save_path": 'distilbert-base-uncased-mean',
-    },
-    {
-        "pooling_strategy": 'min',
-        "pretrained_model_name_or_path": 'distilbert-base-uncased',
-        "model_save_path": 'distilbert-base-uncased-min',
-    },
-    {
-        "pooling_strategy": 'max',
-        "pretrained_model_name_or_path": 'distilbert-base-uncased',
-        "model_save_path": 'distilbert-base-uncased-max',
-    },
-    {
-        "pretrained_model_name_or_path": 'xlnet-base-cased',
-        "model_save_path": 'xlnet-base-cased',
-    },
-    {
-        "pooling_strategy": 'auto',
-        "pretrained_model_name_or_path": 'xlnet-base-cased',
-        "model_save_path": 'xlnet-base-cased-mean',
-    },
-    {
-        "pooling_strategy": 'min',
-        "pretrained_model_name_or_path": 'xlnet-base-cased',
-        "model_save_path": 'xlnet-base-cased-min',
-    },
-    {
-        "pooling_strategy": 'max',
-        "pretrained_model_name_or_path": 'xlnet-base-cased',
-        "model_save_path": 'xlnet-base-cased-max',
-    },
-    {
-        "pooling_strategy": 'mean',
-        "pretrained_model_name_or_path": 'distilbert-base-uncased',
-        "model_save_path": 'distilbert-base-uncased-mean',
-        "layer_index": -2,
-    },
-    {
-        "pooling_strategy": 'mean',
-        "pretrained_model_name_or_path": 'distilbert-base-cased',
-        "model_save_path": 'distilbert-base-cased-mean',
-        "max_length": 100
-    },
-]
 
 
 @pytest.fixture
 def encoder(request, test_metas):
+    request.param["model_save_path"] = (
+        request.param["pretrained_model_name_or_path"].replace(
+            "sentence-transformers/", ""
+        )
+        + f'-{request.param["pooling_strategy"]}'
+    )
     return TransformerTorchEncoder(metas=test_metas, **request.param)
 
 
-@pytest.mark.parametrize('encoder', encoders_parameters, indirect=['encoder'])
+_params_dict = {
+    "pretrained_model_name_or_path": [
+        "sentence-transformers/distilbert-base-nli-stsb-mean-tokens",
+        "sentence-transformers/bert-base-nli-stsb-mean-tokens",
+        "sentence-transformers/xlm-r-100langs-bert-base-nli-stsb-mean-tokens",
+        "sentence-transformers/roberta-base-nli-stsb-mean-tokens",
+        "xlnet-base-cased",
+    ],
+    "pooling_strategy": ["cls", "mean", "max"],
+    "layer_index": [-1, -2, 0],
+}
+
+# Create a cartesian product of the param lists
+encoder_params = [
+    dict(zip(_params_dict.keys(), vals)) for vals in product(*_params_dict.values())
+]
+encoder_params_ind_1 = filter(lambda x: x["layer_index"] == -1, encoder_params)
+
+
+def _assert_params_equal(params_dict: dict, encoder: TransformerTorchEncoder):
+    for key, val in params_dict.items():
+        assert val == getattr(encoder, key)
+
+
+@pytest.mark.parametrize("encoder", encoder_params, indirect=True)
```

Comment:
This will make the test much more readable

Reply:
The problem is, if I do this, then I have to put a statement like
``` python
encoder = Torchencoder(
    param_1=param_1,
    param_2=param_2,
    ...
)
```

in basically every test, as there is no way to use pytest fixtures indirectly for multiple params

Reply:
not sure if is possible but I still prefer this because it makes more explicit than the need to understand how the encoder was built. Plus if a test fails it has in the test name the combination of parameters.